TESTING.md

```diff
@@ -0,0 +1,159 @@
+
+These tests are in the `snapcraft/tests` directory.
+
+### Integration tests
+
+The integration tests are a group of suites that excercise snapcraft as a black box. They are only allowed to set up the environment where snapcraft runs and create files; but for the execution phase of the test they can only run the snapcraft command or one of its subcommands. To verify the results they can check the output printed to the command line, the return value of the snapcraft command, and any files created during the execution.
+
+This suite was split in three: plugins, store and other integration tests. This split is artificial, we made it just because the full suite takes more time than what Travis allows for a single job.
+
+These tests are in the `integration_tests` directory, with the `snapcraft.yamls` and other source files for the tests snaps in `integration_tests/snaps`.
+
+### Snaps tests
+
+The snaps tests is a suite of high-level tests that try to simulate real-world scenarios of a user interacting with snapcraft. They cover the call to snapcraft to generate a snap file from the source files of a fully functional project, the installation of the resulting snap, and the execution of the binaries and services of this snap.
+
+These tests are in the `snaps_tests` directory, with the sources for the test snaps in the `demos` directory.
+
+### Setting up the environment
+
+In order to run these tests suites, first you will need to set up your development environment. Follow the steps in the [Hacking guide](HACKING.md) to install for development.
+
+Then, you'll need a few more dependencies:
+
+    sudo apt install squashfs-tools xdelta3 bzr git mercurial subversion
+
+### Running the tests
+
+To run all the tests execute:
+
+    ./runtests.sh
+
+You can selectively run only one of the suites, and apply a filter:
+
+    ./runtests.sh [static|unit|integration|plugins|store|snaps] [pattern]
+
+Examples:
+
+  * To run only the static tests:
+
+    ```
+    ./runtests.sh static
+    ```
+
+  * To run only the full unit test suite:
+
+    ```
+    ./runtests.sh unit
+    ```
+
+  * To run only the unit tests in the file called `test_init.py`:
+
+    ```
+    ./runtests.sh unit test_init.py
```

Comment:
@jamesbeedy individual files, yes. To run an individual test, you will have to use python3 -m unittest directly.